### ISSUE TITLE
How is "relative depth" defined precisely?

### ISSUE THREAD

**Issue by foamliu** [google/mediapipe#2357]
Thank you very much for the wonderful work! Can you tell us more about how "relative depth" is defined?

**Comment by sgowroji** [comment]
Hi @foamliu, Could you please elaborate your query with more details. Thanks!

**Comment by foamliu** [comment]
![relative_depth](https://user-images.githubusercontent.com/14009435/127603744-a8ba2569-1b8a-4ca9-837c-19cbbfb3ceef.png)
Hello @sgowroji , thanks for quick response, let me explain the query:
Given wrist landmark coordinates(X0, Y0, Z0) in object space(camera coordinate system), arbitary landmark (Xi, Yi, Zi) and camera parameters, we can compute their image points (x0, y0) and (xi, yi) in image plane based on pinhole camera model. However, I( don't know how to calculate relative depth zi with above information. I'm trying to calculate zi using the following formula, is that correct?

![relative_depth_2](https://user-images.githubusercontent.com/14009435/127604489-e959e712-680e-4f17-a796-d258a86464b4.PNG)

**Comment by ahmadyan** [comment]
I'm not sure if this question is related to Objectron. Figure above is definitely not representative of how we compute depth. 
We use homogenous points for projection, so if p=(x, y, z) --> p_2d = (x/z, y/z).

You can read more about this from the book Multiview geometry by Hartly Zisserman.

**Comment by foamliu** [comment]
Thanks ahmadyan. Maybe I didn't make it clear. Homogeneous coordinates are fine. I mean, when you train the MediaPipe Hands 2.5D landmarks prediction model, you feed the model with image and label. For each hand, the annotation information includes the 21x3 2.5 coordinate. For each joint, the coordinates are xi, yi, and zi. The first two are the pixel coordinates of the joint in the image. How is the third one -- zi, the relative depth of the joint, calculated? In the synthetic dataset, you certainly have every landmark coordinates (Xi, Yi, Zi) in the camera coordinate system. You also have camera parameters. How do you calculate 2.5D coordinates from 3D coordinates?

**Comment by foamliu** [comment]
The question is related to MediaPipe Hands, not Objectron. :)

**Comment by foamliu** [comment]
Hello, Ping~ :） @bazarevsky

**Comment by foamliu** [comment]
Hello, Ping~ :） @bazarevsky @sgowroji

**Comment by foamliu** [comment]
Hello, Ping~ :） @bazarevsky @sgowroji

**Comment by bazarevsky** [comment]
To obtain ground truth for BlazePose and Hands we use GHUM model (https://github.com/google-research/google-research/tree/master/ghum). We reconstruct object pose in 3D meter space on the basis of various signals, like 2d keypoint prediction, segmentation mask, etc. Some details related fitting can be fond here (https://blog.tensorflow.org/2021/08/3d-pose-detection-with-mediapipe-blazepose-ghum-tfjs.html). 

However, due to BlazePose/Hand pipeline architecture, they operate on a cropped image, and that mean, for cropped region, we need to predict 3D in a relative manner. That is why, we normalize 3d metric coordinates (obtained from the fitting) so center of coordinates is not a camera, but some subjects keypoint (For pose it would be average of left and right hip. For hand: middle finger first knuckle).
